stan/math/prim/mat/matrix_addons.h

```diff
@@ -0,0 +1,174 @@
+#ifndef STAN_MATH_PRIM_MAT_MATRIX_ADDONS_H
+#define STAN_MATH_PRIM_MAT_MATRIX_ADDONS_H
+
+/**
+ * Reimplements is_fvar without requiring external math headers
+ *
+ * decltype((void)(T::d_)) is a pre C++17 replacement for std::void_t
+ */
+template<class, class = void>
+struct is_fvar : std::false_type
+{ };
+template<class T>
+struct is_fvar<T, decltype((void)(T::d_))> : std::true_type
+{ };
+
+template<typename T>
+using rev_rtn_type = std::conditional_t<std::is_const<typename std::remove_reference<T>::type>::value,
```

Comment:
```suggestion
using rev_rtn_type = std::conditional_t<std::is_const<std::remove_reference_t<T>>::value,
```

Maybe put a todo here so that when we move to C++17 we can use `std::is_const_v<>`